src/core/src/op/fake_convert.cpp

```diff
@@ -0,0 +1,563 @@
+void FakeConvert::validate_and_infer_types() {
+    OV_OP_SCOPE(v13_FakeConvert_validate_and_infer_types);
+    validate_type();
+    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+}
+
+std::shared_ptr<ov::Node> FakeConvert::clone_with_new_inputs(const ov::OutputVector& new_args) const {
+    OV_OP_SCOPE(v13_FakeConvert_clone_with_new_inputs);
+    OPENVINO_ASSERT(new_args.size() == 3, "Incorrect number of new arguments");
+
+    return std::make_shared<FakeConvert>(new_args.at(0),
+                                         new_args.at(1),
+                                         new_args.at(2),
+                                         m_destination_type,
+                                         m_apply_scale);
+}
+
+bool FakeConvert::visit_attributes(ov::AttributeVisitor& visitor) {
+    OV_OP_SCOPE(v13_FakeConvert_visit_attributes);
+    visitor.on_attribute("destination_type", m_destination_type);
+    visitor.on_attribute("apply_scale", m_apply_scale);
+
+    return true;
+}
+
+void FakeConvert::validate_type() const {
+    const auto& valid_types = fake_convert::get_valid_types();
+    OPENVINO_ASSERT(std::find(valid_types.begin(), valid_types.end(), m_destination_type) != valid_types.end(),
+                    "Bad format for f8 conversion type: " + m_destination_type);
+}
+
+// bool FakeConvert::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+//     OV_OP_SCOPE(v13_FakeConvert_evaluate);
+//     OPENVINO_ASSERT(outputs.size() == 1);
+//     OPENVINO_ASSERT(inputs.size() == 3);
+
+//     const auto& data_shape = inputs[0].get_shape();
+//     outputs[0].set_shape(data_shape);
+
+//     return true;
+// }
+
+bool FakeConvert::has_evaluate() const {
+    OV_OP_SCOPE(v13_FakeConvert_has_evaluate);
+    switch (get_input_element_type(0)) {
+    case element::f16:
+    case element::f32:
+        return true;
+    default:
+        return false;
+    }
+}
+
+namespace convert_fp8 {
+namespace {
+void print_tensor(const ov::Tensor& t, std::string s) {
+    std::cout << "Tensor " << s << ": ";
+    auto shape = t.get_shape();
+    int len = shape_size(shape);
+
+    if (t.get_element_type() == ov::element::f16) {
+        auto ptr = static_cast<ov::float16*>(t.data());
+        for (int i = 0; i < len; i++) {
+            std::cout << ptr[i] << " ";
+        }
+    }
+    if (t.get_element_type() == ov::element::f32) {
+        auto ptr = static_cast<float*>(t.data());
+        for (int i = 0; i < len; i++) {
+            std::cout << ptr[i] << " ";
+        }
+    }
+    std::cout << std::endl;
+}
+/// <summary>
+/// emulation of convertation fp16 value to bf8 1s-5e-2m format, Brain Float
+/// </summary>
+/// <typeparam name="T">Every possible type with 16 bit size</typeparam>
+/// <param name="arg"></param>
+/// <param name="out"></param>
+/// <param name="count"></param>
+template <typename T>
+void convertfp16_bf8(const T* const arg, T* out, size_t count, int exp_bits = 5, int mbits = 8) {
```

Comment:
[POC update] Move the reference code to a separate file in the .../openvino/reference/

Reply:
Done